packages/vite-plugin-svelte/src/index.ts

```diff
@@ -40,155 +46,164 @@ export function svelte(inlineOptions?: Partial<Options>): Plugin {
+						e
+					);
+					// this error most likely happens due to non-svelte related importee/importers so swallow it here
+					// in case it really way a svelte library, users will notice anyway. (lib not working due to failed resolve)
 				}
-			} catch (e) {
-				log.debug.once(
-					`error trying to resolve ${importee} from ${importer} via package.json svelte field `,
-					e
-				);
-				// this error most likely happens due to non-svelte related importee/importers so swallow it here
-				// in case it really way a svelte library, users will notice anyway. (lib not working due to failed resolve)
-			}
-		},
+			},
 
-		async transform(code, id, opts) {
-			const ssr = !!opts?.ssr;
-			const svelteRequest = requestParser(id, ssr);
-			if (!svelteRequest || svelteRequest.query.svelte) {
-				return;
-			}
-			let compileData;
-			try {
-				compileData = await compileSvelte(svelteRequest, code, options);
-			} catch (e) {
-				throw toRollupError(e, options);
-			}
-			logCompilerWarnings(compileData.compiled.warnings, options);
-			cache.update(compileData);
-			if (compileData.dependencies?.length && options.server) {
-				compileData.dependencies.forEach((d) => {
-					ensureWatchedFile(options.server!.watcher, d, options.root);
-				});
-			}
-			log.debug(`transform returns compiled js for ${svelteRequest.filename}`);
-			return compileData.compiled.js;
-		},
+			async transform(code, id, opts) {
+				const ssr = !!opts?.ssr;
+				const svelteRequest = requestParser(id, ssr);
+				if (!svelteRequest || svelteRequest.query.svelte) {
+					return;
+				}
+				let compileData;
+				try {
+					compileData = await compileSvelte(svelteRequest, code, options);
+				} catch (e) {
+					throw toRollupError(e, options);
+				}
+				logCompilerWarnings(compileData.compiled.warnings, options);
+				cache.update(compileData);
+				if (compileData.dependencies?.length && options.server) {
+					compileData.dependencies.forEach((d) => {
+						ensureWatchedFile(options.server!.watcher, d, options.root);
+					});
+				}
+				log.debug(`transform returns compiled js for ${svelteRequest.filename}`);
+				return compileData.compiled.js;
+			},
 
-		handleHotUpdate(ctx: HmrContext): void | Promise<Array<ModuleNode> | void> {
-			if (!options.hot || !options.emitCss) {
-				return;
-			}
-			const svelteRequest = requestParser(ctx.file, false, ctx.timestamp);
-			if (svelteRequest) {
-				return handleHotUpdate(compileSvelte, ctx, svelteRequest, cache, options);
+			handleHotUpdate(ctx: HmrContext): void | Promise<Array<ModuleNode> | void> {
+				if (!options.hot || !options.emitCss) {
+					return;
+				}
+				const svelteRequest = requestParser(ctx.file, false, ctx.timestamp);
+				if (svelteRequest) {
+					return handleHotUpdate(compileSvelte, ctx, svelteRequest, cache, options);
+				}
 			}
 		}
-	};
+	];
+	plugins.push(svelteInspector());
+	return plugins.filter(Boolean);
```

Comment:
plugin initializers can decide not to participate by returning undefined, filter falsy